release_notes/v2.2.1.md

```diff
@@ -0,0 +1,160 @@
+What's New in Hyperledger Fabric v2.2.1
+---------------------------------------
+
+Fixes
+-----
+
+**FAB-18163: orderer certificate expiration - TLSHandshakeTimeShift without separate cluster port**
+
+If the TLS certificates of the ordering service nodes expire and are not replaced in time,
+communication between them cannot be established, and it will be impossible to send
+new transactions to the ordering service. To recover from such a scenario, it is possible
+to configure a backwards timeshift that ordering service nodes will utilize for TLS
+handshakes so that transactions can be processed.
+If the cluster service is sharing the orderer’s main gRPC server port,
+configure the new orderer `General.TLS.TLSHandshakeTimeShift` property.
+If using a separate cluster listener port,
+configure the orderer `General.Cluster.TLSHandshakeTimeShift` property.
+
+**FAB-18205: peer CLI certificate expiration - Permit peer CLI to communicate with orderers with expired TLS certs**
+
+The change allows peer CLI to communicate with ordering service nodes with expired TLS certificates
+by setting the `--tlsHandshakeTimeShift` flag to a desired backwards timeshift.
+The change applies to the `peer channel fetch` and `peer channel update` commands to allow
+fetching configuration blocks and updating the channel configuration for orderers with expired TLS certificates.
+
+**FAB-18171: orderer certificate expiration - Disregard certificate validity period in intra-orderer communication**
+
+This change makes the orderer cluster authentication infrastructure
+disregard validity periods when comparing certificates, and only regard public keys.
+With this change, an expiring Raft TLS certificate can be replaced
+with a new certificate that has the same public key, without issuing channel configuration updates.
+
+**FAB-18188: peer and orderer certificate expiration - Log expiration date upon startup**
+
+The future enrollment, tls server, and tls client certificate expiration dates are now logged upon peer and orderer startup.
+
+**peer and orderer HSM - Drain session pool before creating new sessions**
+
+If a bad session was retrieved from the session pool, a new session
+would be immediately created instead of attempting to retrieve another
+session from the pool.
+This change attempts to uses pooled sessions before creating new ones.
+
+**peer and orderer HSM - Add object handle cache to PKCS#11 provider**
+
+With this change, object handles are cached in the PKCS#11 implementation.
+In support of this change, in addition to pooling idle sessions, the
+provider tracks active sessions. If some condition occurs that results
+in all sessions being closed, cached object handles are no longer valid
+so the handle cache is purged.
+
+**FAB-18208: Do not sign gossip message if membership is empty**
+
+This change suppresses the signing of gossip messages if the message will not get
+sent regardless due to an empty gossip membership. The change improves performance,
```

Comment:
Doesn't improve performance, it reduces CPU consumption and stress on HSM

Reply:
Done